tools/SampleAnalyzer/Process/RegionSelection/RegionSelection.h

```diff
@@ -97,26 +108,39 @@ class RegionSelection
   void SetNumberOfCutsAppliedSoFar(MAuint32 NumberOfCutsAppliedSoFar)
     { NumberOfCutsAppliedSoFar_ = NumberOfCutsAppliedSoFar; }
 
+
   // Increment CutFlow (when this region passes a cut)
-  void IncrementCutFlow(MAfloat64 weight)
+  void IncrementCutFlow(MAfloat64 weight, const std::map<MAuint32, MAfloat64> &multiweight)
   {
     cutflow_[NumberOfCutsAppliedSoFar_].Increment(weight);
+	cutflow_[NumberOfCutsAppliedSoFar_].Increment(multiweight);
     NumberOfCutsAppliedSoFar_++;
+
   }
 
   // Add a cut to the CutFlow
   void AddCut(std::string const &CutName)
     { cutflow_.InitCut(CutName); }
 
   /// Getting ready for a new event
-  void InitializeForNewEvent(const MAfloat64 &weight)
+  void InitializeForNewEvent(const MAfloat64 &weight,const std::map<MAuint32, MAfloat64> &multiweight)
```

Comment:
This is not backwards compatible; it breaks all the codes developed in the past! Please split this into two as follows;

```cpp
/// Initializing the new event with a single weight
void InitializeForNewEvent(const MAfloat64 &weight);

/// Initializing the new event with multi-weight
void InitializeForNewEvent(const std::map<MAuint32, MAfloat64> &multiweight);
```

This way, we won't need to modify old implementations individually. You can test old implementations by using the `install PADForSFS` command, which will download a set of implementation codes in `tools/PADForSFS/Build/SampleAnalyzer/User/Analyzer`, and you can see which functions can not be changed. All those codes have to work.